gulpfile.js

```diff
@@ -39,80 +47,130 @@ gulp.task('lint', function() {
-      `${APP_ROOT}js/bootstrap.js`,
-    ])
-    .pipe(concat('initapp.js'))
-    .pipe(gulp.dest(`${DIST_APP_ROOT}js`));
+gulp.task('copy-app', function() {
+  return merge(
+    gulp.src([
+      `${APP_ROOT}**`,
+      // Don't copy documentation files.
+      `!${APP_ROOT}**/*.md`,
+      // Don't copy JS, it will be compiled and copied on the compile step.
+      `!${APP_ROOT}js/**`,
+    ]),
+
+    // Module loader.
+    gulp.src('./node_modules/alameda/alameda.js')
+      .pipe(rename('js/alameda.js')),
+
+    // Components.
+    gulp.src('./node_modules/react/dist/react.min.js')
+      .pipe(rename('js/components/react.js')),
+    gulp.src('./node_modules/react-dom/dist/react-dom.min.js')
+      .pipe(rename('js/components/react-dom.js')),
+    gulp.src('./node_modules/fxos-mvc/dist/mvc.js')
+      .pipe(rename('js/components/mvc.js')),
+
+    // Polyfills.
+    gulp.src('./node_modules/whatwg-fetch/fetch.js')
+      .pipe(rename('js/polyfills/fetch.js')),
+    gulp.src('./node_modules/url-search-params/build/url-search-params.js')
+      .pipe(rename('js/polyfills/url-search-params.js'))
+  )
+  .pipe(gulp.dest(DIST_APP_ROOT));
 });
 
 /**
- * Copy all non-js directory app source/assets/components.
+ * Compiles app source i.e. processes source with Bable and Rollup.
  */
-gulp.task('copy-app', function() {
-  return gulp.src([
-      `${APP_ROOT}**`,
-      `!${APP_ROOT}js/**/*.js`, // do not copy js
-    ])
-    .pipe(gulp.dest(DIST_APP_ROOT));
+gulp.task('compile-app', function() {
+  return gulp
+    .src(`${APP_ROOT}js/*.js`)
+    .pipe(
+      rollup({
+        sourceMap: true,
+        format: 'amd',
+        external: [
+          'components/mvc',
+          'components/react',
+          'components/react-dom',
+        ],
+        plugins: [
+          rollupBabelPlugin({ presets: ['es2015-rollup'] }),
+          rollupIncludePathsPlugin({ extensions: ['.js', '.jsx'] }),
+        ],
+      })
+    )
+    .pipe(sourcemaps.write('.'))
+    .pipe(gulp.dest(`${DIST_APP_ROOT}js`));
 });
 
 /**
- * Apply Babel transform to the JS and JSX files.
+ * Compiles app source for production i.e. processes source with Bable, Rollup
+ * and UglifyJS.
  */
-gulp.task('babel-app', function() {
-  const files = [
-    `${APP_ROOT}js/**/*.{js,jsx}`,
-  ];
-
-  try {
-    return gulp.src(files)
-      .pipe(babel().on('error', function(e) {
-          console.log('error running Babel', e);
-        })
-      )
-      .pipe(gulp.dest(`${DIST_APP_ROOT}js/`));
-  } catch (e) {
-    console.log('Got error in Babel', e);
-  }
+gulp.task('compile-app-production', ['compress-external-modules'], function() {
+  return gulp
+    .src(`${APP_ROOT}js/*.js`)
+    .pipe(
+      rollup({
+        sourceMap: true,
+        format: 'amd',
+        external: [
+          'components/mvc',
+          'components/react',
+          'components/react-dom',
+        ],
+        plugins: [
+          rollupBabelPlugin({ presets: ['es2015-rollup'] }),
```

Comment:
So that looks like a bug in plugin, adding additional settings should have complemented .babelrc, not entirely override it - I'll look how to fix that.